src/readers/src/T2D.h

```diff
@@ -0,0 +1,240 @@
+		spheres_centers_ = SerializableVec<Coords>{}.set_size(count);
+	}
+	float diameter_{0};
+	SerializableVec<Coords> spheres_centers_;
+
+	friend std::istream& operator>>(std::istream& in, Spheres& spheres) {
+		in >> spheres.diameter_;
+		in >> spheres.spheres_centers_;
+		return in;
+	}
+};
+
+struct SpheresGroup {
+	SpheresGroup& init(const std::vector<size_t>& groups_spheres_count) {
+		groups_spheres_count_ = groups_spheres_count;
+		return *this;
+	}
+
+	friend std::istream& operator>>(std::istream& in, SpheresGroup& spheres) {
+		for (auto count : spheres.groups_spheres_count_) {
+			Spheres spheres_lauer{count};
+			in >> spheres_lauer;
+			spheres.spheres_groups_.push_back(spheres_lauer);
+		}
+		return in;
+	}
+	std::vector<size_t> groups_spheres_count_;
+	std::vector<Spheres> spheres_groups_;
+};
+}  // namespace
+
+class T2D {
+ public:
+	size_t layers_count() const { return IST_; }
+
+	const std::vector<float>& net_x() const { return net_x_coords.data(); }
+
+	const std::vector<float>& net_y() const { return net_y_coords.data(); }
+
+	const std::vector<float>& temperatures() const { return temperature_.data(); }
+
+	friend std::istream& operator>>(std::istream& istream, T2D& t2d) {
+		istream >> t2d.program_name_;
+		istream >> t2d.body_size_;
+		istream >> t2d.IST_;
+		istream >> t2d.IH_;
+		istream >> t2d.ID_;
+		istream >> t2d.IB_;
+		istream >> t2d.IV_;
+		istream >> t2d.IP_;
+		istream >> t2d.NH_;
+		istream >> t2d.MH_;
+		istream >> t2d.raw_layers_;
+		istream >> t2d.n_crystals_in_crystal_layer_;
+		istream >> t2d.common_layesr_num_crystals_;
+		istream >> t2d.box_num_;
+		istream >> t2d.spheres_count_.set_size(t2d.IB_);
+		istream >> t2d.common_layesr_num_spheres_.set_size(t2d.IB_);
+		istream >> t2d.spheres_out_layers_.set_size(t2d.IB_);
+		istream >> t2d.body_num_;
+		istream >> t2d.body_params_;
+		istream >> t2d.body_coords_;
+		if (t2d.IH_ > 0)
+			istream >> t2d.ihs_.set_size(t2d.IST_);
+		istream >> t2d.temp_.set_size(t2d.IST_);
+		istream >> t2d.ids_.set_size(t2d.n_crystals_in_crystal_layer_);
+		istream >> t2d.spheres_.init(t2d.spheres_count_.data());
+		istream >> t2d.net_y_coords.set_size(t2d.MH_);
+		istream >> t2d.net_x_coords.set_size(t2d.NH_);
+		istream >> t2d.min_max_layers_coords_.set_size(t2d.IST_ * 2 * 2);
+		istream >> t2d.thicknesses_.set_size(t2d.IST_);
+		istream >> t2d.topology_.set_size(t2d.IST_ * t2d.NH_ * t2d.MH_);
+		istream >> t2d.env_temp_;
+		istream >> t2d.temperature_.set_size(t2d.IST_ * t2d.NH_ * t2d.MH_);
+		return istream;
+	}
+
+ private:
+	std::string program_name_;
+
+	BodySize body_size_;
+
+	size_t IST_;  //– общее число слоев в корпусе,
+	size_t IH_;   // – число однородных слоев,
+	size_t ID_;   // – число слоев кристаллов ИС,
+	size_t IB_;   // – число слоев шариковых выводов,
+	size_t IV_;   // – число слоев межсоединений1
+	size_t IP_;   // – число боксов2
+	size_t NH_;   // – число узлов сетки по оси x,
+	size_t MH_;   // – число узлов сетки по оси y
```

Comment:
1) english
2) are these names same as in doc? If not, please, rename them and make more understandable